src/cmd/services/m3dbnode/config/fs.go

```diff
@@ -78,11 +88,152 @@ type FilesystemConfiguration struct {
 
 	// ForceIndexSummariesMmapMemory forces the mmap that stores the index lookup bytes
 	// to be an anonymous region in memory as opposed to a file-based mmap.
-	ForceIndexSummariesMmapMemory bool `yaml:"force_index_summaries_mmap_memory"`
+	ForceIndexSummariesMmapMemory *bool `yaml:"force_index_summaries_mmap_memory"`
 
 	// ForceBloomFilterMmapMemory forces the mmap that stores the index lookup bytes
 	// to be an anonymous region in memory as opposed to a file-based mmap.
-	ForceBloomFilterMmapMemory bool `yaml:"force_bloom_filter_mmap_memory"`
+	ForceBloomFilterMmapMemory *bool `yaml:"force_bloom_filter_mmap_memory"`
+}
+
+// Validate validates the Filesystem configuration. We use this method to validate
+// fields where the validator package falls short.
+func (f FilesystemConfiguration) Validate() error {
+	if f.WriteBufferSize != nil && *f.WriteBufferSize < 1 {
```

Comment:
Would it be worth setting the defaults in this method so that it's not behind an additional nil check every time you use it?

Reply:
Seems like a little too much. Like if you forget to call validate you don't get default values which is weird. Would rather leave as is

Reply:
Sure, sgtm; I guess you kinda have to run `Validate()` now already to avoid garbage values, but all good :+1: